server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3344,46 +3354,45 @@ public String toString() {
         }
     }
 
-    /**
-     * Updates the shard status in the cluster state
-     *
-     * @param update shard snapshot status update
-     */
-    private void innerUpdateSnapshotState(ShardSnapshotUpdate update, ActionListener<Void> listener) {
+    private void innerUpdateSnapshotState(
+        Snapshot snapshot,
+        ShardId shardId,
+        RepositoryShardId repoShardId,
+        ShardSnapshotStatus updatedState,
+        ActionListener<Void> listener
+    ) {
+        var update = new ShardSnapshotUpdate(snapshot, shardId, repoShardId, updatedState, new ActionListener<>() {
```

Comment:
We tend to prefer `listener.delegateFailure(...)` in this case, so that the listener has a proper `toString()` and asserts that the inner listener doesn't throw where it shouldn't and so on. Not a big deal but still worth doing IMO.